File: myonlinefm/src/main/java/com/sec/myonlinefm/classificationprogram/data/ClassifyRecommendPattern.java
package com.sec.myonlinefm.classificationprogram.data;

import android.graphics.Bitmap;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Created by gaolin on 2018/4/20.
 */

public class ClassifyRecommendPattern implements Serializable{
    private static volatile ClassifyRecommendPattern mInstance = null;

    public static HashMap<Integer, ClassifyRecommend> classifyRecommendMap = new HashMap<>();

    public static Bitmap[] scrollBitmap = new Bitmap[5];

    private ClassifyRecommendPattern() {
        // It shouldn't delete this constructor for the factory method(Singleton)
    }

    public static ClassifyRecommendPattern getInstance() {
        if(mInstance == null) {
            synchronized (ClassifyRecommendPattern.class){
                if (mInstance == null) {
                    mInstance = new ClassifyRecommendPattern();
                }
            }
        }
        return mInstance;
    }

    public void addRecommendMap(int section_id, ClassifyRecommend classifyRecommend) {
        classifyRecommendMap.put(section_id, classifyRecommend);
    }

    public ClassifyRecommend getClassifyRecommendItem(int section_id) {
        return classifyRecommendMap.get(section_id);
    }
}

File: myonlinefm/src/main/java/com/sec/myonlinefm/classificationprogram/data/ClassifyRecommendPattern.kt
package com.sec.myonlinefm.classificationprogram.data

import android.graphics.Bitmap
import java.io.Serializable

/**
 * Created by gaolin on 2018/4/20.
 */
class ClassifyRecommendPattern private constructor() : Serializable {
    fun addRecommendMap(sectionId: Int, classifyRecommend: ClassifyRecommend?) {
        classifyRecommendMap?.put(sectionId, classifyRecommend)
    }

    fun getClassifyRecommendItem(sectionId: Int): ClassifyRecommend? {
        return classifyRecommendMap?.get(sectionId)
    }

    companion object {
        @Volatile
        private var mInstance: ClassifyRecommendPattern? = null
        var classifyRecommendMap: HashMap<Int?, ClassifyRecommend?>? = HashMap()
        var scrollBitmap: Array<Bitmap?>? = arrayOfNulls<Bitmap?>(5)
        fun getInstance(): ClassifyRecommendPattern? {
            if (mInstance == null) {
                synchronized(ClassifyRecommendPattern::class.java) {
                    if (mInstance == null) {
                        mInstance = ClassifyRecommendPattern()
                    }
                }
            }
            return mInstance
        }
    }
}